File: extract.js
const fs = require('fs');
const path = require('path');

function processHtmlFile(htmlPath, cssPath, jsPath, cssHref, jsSrc) {
  let content = fs.readFileSync(htmlPath, 'utf-8');

  // Extract CSS
  const styleMatch = content.match(/<style>([\s\S]*?)<\/style>/);
  if (styleMatch) {
    let cssContent = styleMatch[1];
    cssContent = cssContent.replace(/url\('fondo\.png'\)/g, "url('../img/fondo.png')");
    cssContent = cssContent.replace(/url\('portada\.jpeg'\)/g, "url('../img/portada.jpeg')");
    cssContent = cssContent.replace(/url\('historia_barber\.jpg'\)/g, "url('../img/historia_barber.jpg')");
    
    fs.mkdirSync(path.dirname(cssPath), { recursive: true });
    fs.writeFileSync(cssPath, cssContent.trim() + '\n', 'utf-8');
    
    content = content.replace(styleMatch[0], `<link rel="stylesheet" href="${cssHref}">`);
  }

  // Extract JS
  const scriptMatch = content.match(/<script>([\s\S]*?)<\/script>/);
  if (scriptMatch) {
    let jsContent = scriptMatch[1];
    
    fs.mkdirSync(path.dirname(jsPath), { recursive: true });
    fs.writeFileSync(jsPath, jsContent.trim() + '\n', 'utf-8');
    
    content = content.replace(scriptMatch[0], `<script src="${jsSrc}"></script>`);
  }

  // Update HTML images
  content = content.replace(/src="logo\.png"/g, 'src="img/logo.png"');
  content = content.replace(/src="letrero\.png"/g, 'src="img/letrero.png"');
  content = content.replace(/src="portada\.jpeg"/g, 'src="img/portada.jpeg"');
  content = content.replace(/src="historia_barber\.jpg"/g, 'src="img/historia_barber.jpg"');
  content = content.replace(/src="(gallery-\d+\.jpg)"/g, 'src="img/$1"');

  fs.writeFileSync(htmlPath, content, 'utf-8');
}

processHtmlFile('index.html', 'css/styles.css', 'js/main.js', 'css/styles.css', 'js/main.js');
processHtmlFile('historia.html', 'css/historia.css', 'js/historia.js', 'css/historia.css', 'js/historia.js');
console.log('Done!');

File: extract.py
import re
import os

def process_html_file(html_path, css_path, js_path, css_href, js_src):
    with open(html_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Extract and replace style
    style_pattern = re.compile(r'<style>(.*?)</style>', re.DOTALL)
    style_match = style_pattern.search(content)
    if style_match:
        css_content = style_match.group(1)
        # Update urls in CSS
        css_content = css_content.replace('url(\'fondo.png\')', 'url(\'../img/fondo.png\')')
        css_content = css_content.replace('url(\'portada.jpeg\')', 'url(\'../img/portada.jpeg\')')
        css_content = css_content.replace('url(\'historia_barber.jpg\')', 'url(\'../img/historia_barber.jpg\')')
        
        os.makedirs(os.path.dirname(css_path), exist_ok=True)
        with open(css_path, 'w', encoding='utf-8') as f:
            f.write(css_content.strip() + '\n')
            
        content = content[:style_match.start()] + f'<link rel=\"stylesheet\" href=\"{css_href}\">' + content[style_match.end():]

    # Extract and replace script
    script_pattern = re.compile(r'<script>(.*?)</script>', re.DOTALL)
    script_match = script_pattern.search(content)
    if script_match:
        js_content = script_match.group(1)
        
        os.makedirs(os.path.dirname(js_path), exist_ok=True)
        with open(js_path, 'w', encoding='utf-8') as f:
            f.write(js_content.strip() + '\n')
            
        content = content[:script_match.start()] + f'<script src=\"{js_src}\"></script>' + content[script_match.end():]

    # Replace image src in HTML
    content = re.sub(r'src=\"logo\.png\"', 'src=\"img/logo.png\"', content)
    content = re.sub(r'src=\"letrero\.png\"', 'src=\"img/letrero.png\"', content)
    content = re.sub(r'src=\"portada\.jpeg\"', 'src=\"img/portada.jpeg\"', content)
    content = re.sub(r'src=\"historia_barber\.jpg\"', 'src=\"img/historia_barber.jpg\"', content)
    # Also handle gallery images: gallery-1.jpg, etc.
    content = re.sub(r'src=\"(gallery-\d+\.jpg)\"', r'src=\"img/\1\"', content)

    with open(html_path, 'w', encoding='utf-8') as f:
        f.write(content)

process_html_file('index.html', 'css/styles.css', 'js/main.js', 'css/styles.css', 'js/main.js')
process_html_file('historia.html', 'css/historia.css', 'js/historia.js', 'css/historia.css', 'js/historia.js')
print('Extraction and replacement complete.')
